src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1361,6 +1362,24 @@ public static bool IsFileTypeOrUsesFileTypes(this TypeSymbol type)
             return foundType is not null;
         }
 
+        internal static string? AssociatedFileIdentifier(this NamedTypeSymbol type)
+        {
+            if (type is not SourceMemberContainerTypeSymbol { AssociatedSyntaxTree: SyntaxTree tree })
+            {
+                return null;
+            }
+            var ordinal = type.DeclaringCompilation.GetSyntaxTreeOrdinal(tree);
+            return GeneratedNames.MakeFileIdentifier(tree.FilePath, ordinal);
+        }
+
+        internal static string GetDisplayFileName(this SyntaxTree tree)
```

Comment:
>GetDisplayFileName [](http://example.com/codeflow?start=31&length=18)

It doesn't look like this helper belongs in this file, "TypeSymbolExtensions.cs"

Reply:
Moved to local function.